src/editors/TestGenerationPanel.ts

```diff
@@ -94,6 +95,8 @@ export class TestGenerationPanel {
       TestGenerationPanel.currentPanel._testRecordNavigation = true;
     }
 
+    TestGenerationPanel.currentPanel.loadAppNames();
+
```

Comment:
Every time I load a test record from the history list, the same list of apps needs to be fetched...is that intended?

Reply:
It was intentional, I didn't want users to be confused why when they reload the screen, even for history, the app's don't match what's in app management. For example, if they load a history instance and realize the app has been deleted, they might re-upload their app and expect to see the app name list reflect that when re-opening the history instance. But if the downsides of calling that API isn't worth that scenario, I can only have it load when it isn't loading history.